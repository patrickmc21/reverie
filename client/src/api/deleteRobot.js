const deleteRobot = async (id) => {
  const url = `/api/v1/hosts/${id}`;
  const options = {
    method: 'DELETE'
  };

  try {
    const response = await fetch(url, options);
    const data = await response.json();
    return data;
  } catch (error) {
    throw error
  }
};

export default deleteRobot;