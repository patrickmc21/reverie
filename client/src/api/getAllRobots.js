const getAllRobots = () => {

};

export default getAllRobots;