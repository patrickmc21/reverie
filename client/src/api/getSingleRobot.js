const getSingleRobot = async (id) => {
  const url = `/api/v1/hosts/${id}`;

  try {
    const response = await fetch(url);
    const robot = await response.json();
    return robot;
  } catch (error) {
    return error;
  }
};

export default getSingleRobot;