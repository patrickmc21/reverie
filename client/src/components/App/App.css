.App-header {
  background-color: #222;
  height: 120px;
  padding: 20px 40px;
  box-sizing: border-box;
}

.App-title {
  float: left;
  margin: 0;
  color: #D9D9D9;
  font-family: 'Open Sans', sans-serif;
  font-size: 48px;
  font-weight: 300;
}

.header--sub {
  color: #000000;
  font-weight: 700;
}

.user-name {
  float: right;
  color: #E0E0E0;
  font-family: 'Roboto', sans-serif;
}

main {
  padding-top: 40px;
  background-image: url('../../assets/background-main.jpg');
  background-size: cover;
  background-repeat: no-repeat;
}

/*
font-family: 'Open Sans', sans-serif;
font-family: 'Roboto', sans-serif;*/