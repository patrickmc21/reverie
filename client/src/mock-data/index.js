export const mockRobots = [
  {
    id: 1,
    date_added: '5/21/2018',
    first_active: '5/21/2018',
    current_name: "R2-D2",
    height: 3.25,
    weight: 200,
    intelligence_metric: 18
  },
  {
    id: 2,
    date_added: '5/20/2018',
    first_active: '5/20/2018',
    current_name: "Bender",
    height: 6.1,
    weight: 250,
    intelligence_metric: 14
  }
];

export const newRobot = {
  date_added: '5/23/2018',
  first_active: '5/23/2018',
  current_name: "T-1000",
  height: 6.5,
  weight: 400,
  intelligence_metric: 19
};

export const updatedRobot = {
  id: 2,
  date_added: '5/23/2018',
  first_active: '5/23/2018',
  current_name: "T-1000",
  height: 6.5,
  weight: 400,
  intelligence_metric: 19
};

