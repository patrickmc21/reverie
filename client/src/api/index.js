import deleteRobot from './deleteRobot';
import getAllRobots from './getAllRobots';
import getSingleRobot from './getSingleRobot';
import postRobot from './postRobot';
import updateRobot from './updateRobot';

const api = {deleteRobot, getAllRobots, getSingleRobot, postRobot, updateRobot};

export default api;