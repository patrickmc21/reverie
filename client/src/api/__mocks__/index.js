import { mockRobots, newRobot, updatedRobot } from '../../mock-data';

const api = {
  deleteRobot: jest.fn().mockImplementation((id) => {
    if (!id) {
      return Promise.reject({message: 'Failed Fetch'});
    }
  }), 
  getAllRobots: jest.fn().mockImplementation(() => {
    return Promise.resolve(mockRobots);
  }), 
  getSingleRobot: jest.fn().mockImplementation((id) => {
    if (id) {
      return Promise.resolve(mockRobots[0]);
    } else {
      return Promise.reject({message: 'Failed Fetch'});
    }
  }), 
  postRobot: jest.fn().mockImplementation((robot) => {
    if (robot) {
      return Promise.resolve(newRobot);
    } else {
      return Promise.reject({message: 'Failed Fetch'});
    }
  }), 
  updateRobot: jest.fn().mockImplementation((robot) => {
    if (robot) {
      return Promise.resolve(updatedRobot);
    } else {
      return Promise.reject({message: 'Failed Fetch'});
    }
  })
};

export default api;