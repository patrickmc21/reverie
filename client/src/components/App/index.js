import React, { Component } from 'react';
import api from '../../api';
import RobotContainer from '../RobotContainer';
import RobotForm from '../RobotForm';
import './App.css';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      robots: [],
      showRobotForm: false,
      editing: null,
      errorStatus: ''
    };
  }

  async componentDidMount() {
    const robots = await api.getAllRobots();
    this.setState({ robots });
  }

  addRobot = async (newRobot) => {
    try {
      const robot = await api.postRobot(newRobot);
      const robots = [...this.state.robots, robot];
      this.setState({ robots, showRobotForm: false });
    } catch (error) {
      this.setState({ errorStatus: error });
    }
  }

  editRobot = async (changedRobot) => {
    try {
      const updatedRobot = await api.updateRobot(changedRobot);
      const robots = this.state.robots.map(robot => {
        if (robot.id === changedRobot.id) {
          return updatedRobot;
        } else {
          return robot
        }
      });
      this.setState({ robots, showRobotForm: false, editing: null });
    } catch (error) {
      this.setState({ errorStatus: error });
    }
  }

  removeRobot = async (id) => {
    try {
      await api.deleteRobot(id);
      const robots = this.state.robots.filter(robot => robot.id !== id);
      this.setState({ robots });
    } catch (error) {
      this.setState({ errorStatus: error });
    }
  }

  triggerForm = (robot) => {
    if (robot) {
      this.setState({showRobotForm: true, editing: robot})
    } else {
      this.setState({showRobotForm: true});
    }
  }

  render() {
    const { robots, showRobotForm, editing } = this.state;
    const { addRobot, editRobot, removeRobot, triggerForm } = this;
    return (
      <div className="App">
        <header className="App-header">
          <h1 
            className="App-title">
            robot
            <span className='header--sub'>
              Portal
            </span>
          </h1>
          <h2 className='user-name'>Welcome, Dr. Robert Ford!</h2>
        </header>
        <main>
          <RobotContainer 
            robots={robots}
            triggerForm={triggerForm}
            removeRobot={removeRobot}
          />
          { showRobotForm && 
            <RobotForm 
              editRobot={editRobot} 
              addRobot={addRobot}
              editing={editing}/> }
        </main>
      </div>
    );
  }
}

export default App;
