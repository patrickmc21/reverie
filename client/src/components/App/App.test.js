import React from 'react';
import { shallow } from 'enzyme';
import api from '../../api';
import App from './index';
import { mockRobots, newRobot, updatedRobot } from '../../mock-data';

jest.mock('../../api');

describe('App', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />, {disableLifecycleMethods: true});
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot when form is active', () => {
    wrapper.setState({showRobotForm: true});
    expect(wrapper).toMatchSnapshot();
  });

  it('should call fetchAllRobots on mount', () => {
    const spy = jest.spyOn(wrapper.instance(), 'fetchAllRobots');
    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalledTimes(1);
  });

  it('should not fetch robots on mount if fetch has happened', () => {
    const spy = jest.spyOn(wrapper.instance(), 'fetchAllRobots');
    wrapper.setState({madeInitialCall: true});
    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalledTimes(0);
  });

  it('should call api getAllRobots in fetchAllRobots', () => {
    wrapper.instance().fetchAllRobots();
    expect(api.getAllRobots).toHaveBeenCalled();
  });

  it('should call api postRobot in addRobot method', () => {
    const expected = newRobot;
    wrapper.instance().addRobot(newRobot);
    expect(api.postRobot).toHaveBeenCalledWith(expected);
  });

  it('should update robot state on addRobot', async () => {
    const expected = [newRobot];
    await wrapper.instance().addRobot(newRobot);
    expect(wrapper.state('robots')).toEqual(expected);
  });

  it('should change errorStatus on failed fetch', async () => {
    const expected = 'Failed Fetch';
    await wrapper.instance().addRobot();
    expect(wrapper.state('errorStatus')).toEqual(expected);
  });

  it('should call api updateRobot in editRobot method', () => {
    const expected = updatedRobot;
    wrapper.instance().editRobot(updatedRobot);
    expect(api.updateRobot).toHaveBeenCalledWith(expected);
  });

  it('should update state on editRobot', async () => {
    const expected = updatedRobot;
    wrapper.setState({robots: mockRobots});
    await wrapper.instance().editRobot(updatedRobot);
    expect(wrapper.state('robots')[1]).toEqual(expected);
  });

  it('should change errorStatus on failed fetch', async () => {
    const expected = 'Failed Fetch';
    await wrapper.instance().editRobot();
    expect(wrapper.state('errorStatus')).toEqual(expected);
  });

  it('should call api deleteRobot in removeRobot method', () => {
    const expected = 1;
    wrapper.instance().removeRobot(1);
    expect(api.deleteRobot).toHaveBeenCalledWith(expected);
  });

  it('should update state on removeRobot', async () => {
    const expected = [mockRobots[1]];
    wrapper.setState({robots: mockRobots});
    await wrapper.instance().removeRobot(1);
    expect(wrapper.state('robots')).toEqual(expected);
  });

  it('should change errorStatus on failed fetch', async () => {
    const expected = 'Failed Fetch';
    await wrapper.instance().removeRobot();
    expect(wrapper.state('errorStatus')).toEqual(expected);
  });

  it('should change showRobotForm to true on triggerForm', () => {
    const expected = true;
    wrapper.instance().triggerForm();
    expect(wrapper.state('showRobotForm')).toEqual(expected);
  });

  it('should change editing state to robot object if robot provided', () => {
    const expected = newRobot;
    wrapper.instance().triggerForm(newRobot);
    expect(wrapper.state('editing')).toEqual(expected);
  });
});

