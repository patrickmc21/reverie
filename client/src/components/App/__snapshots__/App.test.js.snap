// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot when form is active 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <h1
                    className="App-title"
          >
                    robot
                    <span
                              className="header--sub"
                    >
                              Portal
                    </span>
          </h1>
          <h2
                    className="user-name"
          >
                    Welcome, Dr. Robert Ford!
          </h2>
</header>,
        <main>
          <RobotContainer
                    removeRobot={[Function]}
                    robots={Array []}
                    triggerForm={[Function]}
          />
          <RobotForm
                    addRobot={[Function]}
                    editRobot={[Function]}
                    editing={null}
          />
</main>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="App-title"
>
              robot
              <span
                            className="header--sub"
              >
                            Portal
              </span>
</h1>,
            <h2
              className="user-name"
>
              Welcome, Dr. Robert Ford!
</h2>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "robot",
                <span
                  className="header--sub"
>
                  Portal
</span>,
              ],
              "className": "App-title",
            },
            "ref": null,
            "rendered": Array [
              "robot",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Portal",
                  "className": "header--sub",
                },
                "ref": null,
                "rendered": "Portal",
                "type": "span",
              },
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome, Dr. Robert Ford!",
              "className": "user-name",
            },
            "ref": null,
            "rendered": "Welcome, Dr. Robert Ford!",
            "type": "h2",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <RobotContainer
              removeRobot={[Function]}
              robots={Array []}
              triggerForm={[Function]}
/>,
            <RobotForm
              addRobot={[Function]}
              editRobot={[Function]}
              editing={null}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "removeRobot": [Function],
              "robots": Array [],
              "triggerForm": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "addRobot": [Function],
              "editRobot": [Function],
              "editing": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "main",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <h1
                        className="App-title"
            >
                        robot
                        <span
                                    className="header--sub"
                        >
                                    Portal
                        </span>
            </h1>
            <h2
                        className="user-name"
            >
                        Welcome, Dr. Robert Ford!
            </h2>
</header>,
          <main>
            <RobotContainer
                        removeRobot={[Function]}
                        robots={Array []}
                        triggerForm={[Function]}
            />
            <RobotForm
                        addRobot={[Function]}
                        editRobot={[Function]}
                        editing={null}
            />
</main>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="App-title"
>
                robot
                <span
                                className="header--sub"
                >
                                Portal
                </span>
</h1>,
              <h2
                className="user-name"
>
                Welcome, Dr. Robert Ford!
</h2>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "robot",
                  <span
                    className="header--sub"
>
                    Portal
</span>,
                ],
                "className": "App-title",
              },
              "ref": null,
              "rendered": Array [
                "robot",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Portal",
                    "className": "header--sub",
                  },
                  "ref": null,
                  "rendered": "Portal",
                  "type": "span",
                },
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome, Dr. Robert Ford!",
                "className": "user-name",
              },
              "ref": null,
              "rendered": "Welcome, Dr. Robert Ford!",
              "type": "h2",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <RobotContainer
                removeRobot={[Function]}
                robots={Array []}
                triggerForm={[Function]}
/>,
              <RobotForm
                addRobot={[Function]}
                editRobot={[Function]}
                editing={null}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "removeRobot": [Function],
                "robots": Array [],
                "triggerForm": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "addRobot": [Function],
                "editRobot": [Function],
                "editing": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "main",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <h1
                    className="App-title"
          >
                    robot
                    <span
                              className="header--sub"
                    >
                              Portal
                    </span>
          </h1>
          <h2
                    className="user-name"
          >
                    Welcome, Dr. Robert Ford!
          </h2>
</header>,
        <main>
          <RobotContainer
                    removeRobot={[Function]}
                    robots={Array []}
                    triggerForm={[Function]}
          />
</main>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="App-title"
>
              robot
              <span
                            className="header--sub"
              >
                            Portal
              </span>
</h1>,
            <h2
              className="user-name"
>
              Welcome, Dr. Robert Ford!
</h2>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "robot",
                <span
                  className="header--sub"
>
                  Portal
</span>,
              ],
              "className": "App-title",
            },
            "ref": null,
            "rendered": Array [
              "robot",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Portal",
                  "className": "header--sub",
                },
                "ref": null,
                "rendered": "Portal",
                "type": "span",
              },
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome, Dr. Robert Ford!",
              "className": "user-name",
            },
            "ref": null,
            "rendered": "Welcome, Dr. Robert Ford!",
            "type": "h2",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <RobotContainer
              removeRobot={[Function]}
              robots={Array []}
              triggerForm={[Function]}
/>,
            false,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "removeRobot": [Function],
              "robots": Array [],
              "triggerForm": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
        ],
        "type": "main",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <h1
                        className="App-title"
            >
                        robot
                        <span
                                    className="header--sub"
                        >
                                    Portal
                        </span>
            </h1>
            <h2
                        className="user-name"
            >
                        Welcome, Dr. Robert Ford!
            </h2>
</header>,
          <main>
            <RobotContainer
                        removeRobot={[Function]}
                        robots={Array []}
                        triggerForm={[Function]}
            />
</main>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="App-title"
>
                robot
                <span
                                className="header--sub"
                >
                                Portal
                </span>
</h1>,
              <h2
                className="user-name"
>
                Welcome, Dr. Robert Ford!
</h2>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "robot",
                  <span
                    className="header--sub"
>
                    Portal
</span>,
                ],
                "className": "App-title",
              },
              "ref": null,
              "rendered": Array [
                "robot",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Portal",
                    "className": "header--sub",
                  },
                  "ref": null,
                  "rendered": "Portal",
                  "type": "span",
                },
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome, Dr. Robert Ford!",
                "className": "user-name",
              },
              "ref": null,
              "rendered": "Welcome, Dr. Robert Ford!",
              "type": "h2",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <RobotContainer
                removeRobot={[Function]}
                robots={Array []}
                triggerForm={[Function]}
/>,
              false,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "removeRobot": [Function],
                "robots": Array [],
                "triggerForm": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
          ],
          "type": "main",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
